lms/djangoapps/grades/models.py

```diff
@@ -0,0 +1,46 @@
+from django.contrib.auth.models import User
+from django.db import models
+from model_utils.models import TimeStampedModel
+
+from xmodule_django.models import BlockTypeKeyField, CourseKeyField, UsageKeyField
+
+class PersistentSubsectionGradeModel(TimeStampedModel):
+    """
+    A django model tracking persistent grades at the subsection level.
+
+    TODO: intended query patterns are fluid at the moment, document them here as they're firmed up.
+    """
+
+    class Meta(object):
+        index_together = [
+            # TODO: nail down indices as we flesh out the API layer
+        ]
+
+        unique_together = (('user_id', 'course_id', 'block_type', 'block_id'))
+
+
+    subtree_edited_date = models.DateTimeField('last content edit timestamp')
+    user = models.ForeignKey(User)  # CAUTION: FK may not exist in the future
+    earned_grade = models.IntegerField()
+    max_grade = models.IntegerField()
+    visible_blocks = models.ForeignKey(BlockRecord)
+
+    # TODO: Make sure this matches up w/ what Nimisha's exposing in her PR
+    course_version = models.CharField('guid of latest course version', max_length=255)
+
+    # These 3 are essentially a deconstructed UsageKey
+    course_id = CourseKeyField(max_length=255)
+    block_type = BlockTypeKeyField(max_length=255)
+    block_id = models.CharField(max_length=255)  # TODO: CharField correct here?
```

Comment:
I wouldn't worry too much about the extra space. Course info might add something like 20 bytes, which even at a billion rows isn't all that much (and probably compresses out to be even less of a factor). I don't think it's worth even the small amount of extra complexity.